### DISCUSSION TITLE
Did PowerShell replace the old DOS Batch files?

### DISCUSSION THREAD

**Discussion by tcjohnson71** [PowerShell/PowerShell#17596] [Q&A]
I don't know where else to post this question.  So, are Batch Files gone or replaced by PowerShell?

Thanks,

TCJ

**Comment by vexx32** [comment]
Batch files still exist (though rarely I guess?) and are still usable on basically every version of Windows, as far as I know.

**Reply by tcjohnson71** [reply]
Ok - thanks for the answer.

**Reply by rhubarb-geek-nz** [reply]
You could say that Windows batch files processed by cmd.exe replaced the old DOS batch files processed by command.com.

I suggest that batch (*.bat or *.cmd) files are still very common and will not be going away soon. For example even Visual Studio 2022 still uses them to set up the command line environments. Many open source software use them ( eg gradle, maven, tomcat etc ). 
Batch files are still very efficient at what they do and their start up time is faster than launching another copy of pwsh. 

We have over 40 years of batch files in existence, they won't disappear.
